shell/lib/shell.sh

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env bash
+# Provides generic shell helper functions
+
+DEVBASE_CACHED_BINARY_STORAGE_PATH="$HOME/.outreach/.cache/devbase"
+
+# retry calls a given command (must be wrapped in quotes)
+# syntax: retry <interval> <maxRetries> <command> [args...]
+retry() {
+  local interval="$1"
+  local maxRetries="$2"
+  local command="$3"
+
+  # remove interval+command from the argument stack
+  # so we can send it to the command later
+  shift
+  shift
+  shift
+
+  local exitCode=0
+  for i in $(seq 1 "$maxRetries"); do
+    if [[ $i -gt 1 ]]; then
+      echo "RETRYING: $command ($i/$maxRetries)"
+    fi
+
+    # execute command, if succeeds break out of loop and
+    # and reset exitCode
+    "$command" "$@" && exitCode=0 && break
+
+    # preserve the exit code
+    exitCode=$?
+
+    # try again after x time
+    sleep "$interval"
```

Comment:
could be nice to do sleep interval + (i*3) or something to have a little bit of an exponential backoff

Reply:
"exponential" is not being used literally here

Reply:
yeah, but i don't think we need it rn, but happy to add it later if this actually becomes an issue (I suspect this'll only ever really trigger once)